Bottom: c89bca1000112fd468411b0a3b6d4d7b4146e154
Top:    69bc8a41926b3ef0a1a4e6f2d8bb8cd8b26ea1d5
Author: Jason Wang <jasowang@redhat.com>
Date:   2015-09-17 13:57:00 +0800

Refresh of timer-completion

---

diff --git a/drivers/net/virtio_net.c b/drivers/net/virtio_net.c
index f2793f2..0388969 100644
--- a/drivers/net/virtio_net.c
+++ b/drivers/net/virtio_net.c
@@ -1544,8 +1544,8 @@ static enum hrtimer_restart virtnet_complete_tx(struct hrtimer *timer)
 	struct netdev_queue *txq = netdev_get_tx_queue(vi->dev,
 						vq2txq(sq->vq));
 
-	__netif_tx_lock_bh(txq);
-//	free_old_xmit_skbs(sq);
+	__netif_tx_lock(txq, smp_processor_id());
+	free_old_xmit_skbs(sq);
 //	virtqueue_foreach_buf(sq->vq, virtnet_orphan_skb);
 	__netif_tx_unlock(txq);
